File: contracts/wasm/timestamp/go/timestamp/timestamp.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package timestamp

import "github.com/iotaledger/wasp/packages/wasmvm/wasmlib/go/wasmlib"

func funcNow(ctx wasmlib.ScFuncContext, f *NowContext) {
}

func viewGetTimestamp(ctx wasmlib.ScViewContext, f *GetTimestampContext) {
	f.Results.Timestamp().SetValue(ctx.Timestamp())
}

File: contracts/wasm/timestamp/rs/timestamp/src/timestamp.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;

pub fn func_now(_ctx: &ScFuncContext, _f: &NowContext) {
}

pub fn view_get_timestamp(ctx: &ScViewContext, f: &GetTimestampContext) {
    f.results.timestamp().set_value(ctx.timestamp());
}
